test/sqlancer/reducer/TestASTBasedReducer.java

```diff
@@ -111,19 +151,51 @@ void testConstantVar() throws Exception {
     }
 
     @Test
-    void testFunction() throws Exception {
+    void testSubSelects() throws Exception {
+        TestEnvironment env = TestEnvironment.getASTBasedReducerEnv();
+
+        String[] queriesStrs = {
+                // "SELECT AVG(sum_column1) FROM (SELECT SUM(column1) AS sum_column1 FROM t1 GROUP BY column1 LIMIT 32
+                // OFFSET 128) AS t1;",
+                "WITH cte1 AS (SELECT a, b FROM table1), cte2 AS (SELECT c, d FROM table2) SELECT b, d FROM cte1 JOIN cte2 WHERE cte1.a = cte2.c" };
+        env.setInitialStatementsFromStrings(List.of(queriesStrs));
+        env.setBugInducingCondition(statements -> {
+            String queriesString = TestEnvironment.getQueriesString(statements);
+            try {
+                for (Query<?> s : statements) {
+                    CCJSqlParserUtil.parse(s.getQueryString());
+                }
+            } catch (JSQLParserException e) {
+                return false;
+            }
+            return queriesString.contains("cte2");
+        });
+        env.runReduce();
+        List<Query<?>> reducedResult = env.getReducedStatements();
+        System.out.println(Arrays.toString(queriesStrs));
```

Comment:
For test cases, it might be better if we compare against the expected result here? we can still update it if the reducer becomes more effective in the future.